lib/quandl/api_config.rb

```diff
@@ -4,29 +4,36 @@ class ApiConfig
     API_BASE_THREAD_KEY = 'quandl_api_base'
     API_VERSION_THREAD_KEY = 'quandl_api_version_key'
 
+    @api_key = nil
+    @api_base = nil
+    @api_version = nil
+
     class << self
       def api_key=(api_key)
+        @api_key = api_key if @api_key.nil?
```

Comment:
looks like once the api_key is set it can never be set to another value. not sure if that's what we want?